### PR TITLE
adding update checker test env variable

### PR DESCRIPTION
**Pull request by aronchick**

### CODE REVIEW

File: pkg/version/update.go [hunk 1]
```diff
@@ -61,6 +61,10 @@ func CheckForUpdate(
 	}
 	q.Set("clientID", clientID)
 	q.Set("InstallationID", InstallationID)
+	if os.Getenv("BACALHAU_UPDATE_CHECKER_TEST") != "" {
+		q.Set("bacalhau_update_checker_test", "true")
+	}
+
```
Comment (wdbaruni) [inline pkg/version/update.go:67]: What will this do? Might be worth adding a comment here

Comment (aronchick) [reply pkg/version/update.go:67]: will do

File: pkg/version/update.go [hunk 2]
```diff
@@ -61,6 +61,14 @@ func CheckForUpdate(
 	}
 	q.Set("clientID", clientID)
 	q.Set("InstallationID", InstallationID)
+
+	// The BACALHAU_UPDATE_CHECKER_TEST is an env variable a user can set so that we can track
+	// when the binary is being run by a non-user, to enable easier filtering of queries
+	// to their update server for internal/CI.
```
Comment (frrist) [inline pkg/version/update.go:67]: Can I assume this has some relation to: https://github.com/bacalhau-project/getbacalhauorg/pull/29 ?

> when the binary is being run by a non-user

What does this mean? Like a linux user or a person who uses bacalhau?

Comment (aronchick) [reply pkg/version/update.go:67]: person who uses / deploys / etc bacalhau.

Basically just let users set (if they want) a key indicating this run is "internal".

**Pull request merged**